data/src/main/scala/ch.epfl.scala.index.data/github/GithubDownload.scala

```diff
@@ -246,49 +257,109 @@ class GithubDownload(paths: DataPaths,
   }
 
   /**
-    * Process the downloaded data from GraphQL API
+    * Process the downloaded data from repository info
     *
     * @param repo the current repo
     * @param response the response
     * @return
     */
+  private def processCommunityProfileResponse(repo: GithubRepo,
+                                  response: WSResponse): Unit = {
+
+    if (200 == response.status) {
+
+      saveJson(githubRepoCommunityProfilePath(paths, repo), repo, response.body)
+    }
+
+    ()
+  }
+
+  /**
+    * Process the downloaded data from GraphQL API
+    * @param isRetry whether or not the call to this function is a retry
+    * @param repo the current repo
+    * @param response the response
+    * @return
+    */
   private def processTopicsResponse(isRetry: Boolean)(
       repo: GithubRepo,
       response: WSResponse): Unit = {
 
+      val errorOccurred = checkGraphqlError[GithubRepo](repo, response, isRetry, topicQuery, processTopicsResponse)
+
+      if (!errorOccurred) {
+        saveJson(githubRepoTopicsPath(paths, repo), repo, response.body)
+      }
+
+      ()
+  }
+
+  /**
+    * Process the downloaded issues from GraphQL API
+    *
+    * @param isRetry whether or not the call to this function is a retry
+    * @param project the current project
+    * @param response the response
+    * @return
+    */
+  private def processIssuesResponse(isRetry: Boolean)(
+      project: Project,
+      response: WSResponse): Unit = {
+
+    val errorOccurred = checkGraphqlError[Project](project, response, isRetry, issuesQuery, processIssuesResponse)
+
+    if (!errorOccurred) {
+      saveJson(githubRepoIssuesPath(paths, project.githubRepo), project.githubRepo, response.body)
+    }
+
+    ()
```

Comment:
[Done](https://github.com/scalacenter/scaladex/pull/432/commits/0d0e08ce6dbd290c746345aa8c56493a424dcd08#diff-e03c541cf1bd7ec0322a9a6571160beb), updated it to use `Try`